test/unit/lib/tasks/asset_manager_test.rb

```diff
@@ -0,0 +1,168 @@
+require "rake"
+require "test_helper"
+require "gds_api/test_helpers/asset_manager"
+
+class AssetManagerTest < ActiveSupport::TestCase
+  extend Minitest::Spec::DSL
+  include GdsApi::TestHelpers::AssetManager
+
+  let(:task) { Rake::Task["asset_manager:report_deleted_assets"] }
+
+  teardown do
+    task.reenable
+  end
+
+  context "for an asset attached only to an unpublished edition" do
+    let(:edition) { create(:unpublished_publication) }
+    let(:attachment) { create(:file_attachment_with_asset, attachable: edition) }
+
+    context "when asset is not deleted from asset manager" do
+      before do
+        attachment.attachment_data.assets.each do |asset|
+          stub_asset_manager_has_an_asset(asset.asset_manager_id, { deleted: false, file_url: asset.filename })
+        end
+      end
+
+      test "it should include the attachment in the report" do
+        assert_output(/^#{attachment.attachment_data.assets.first.filename},#{edition.public_url},#{edition.state}$/) { task.invoke }
+      end
+    end
+
+    context "when asset is deleted from asset manager" do
+      before do
+        attachment.attachment_data.assets.each do |asset|
+          stub_asset_manager_has_an_asset(asset.asset_manager_id, { deleted: true, file_url: asset.filename })
+        end
+      end
+
+      test "it should not include the attachment in the report" do
+        assert_output(/^(?!#{attachment.attachment_data.assets.first.filename},#{edition.public_url},#{edition.state}).*$/) { task.invoke }
+      end
+    end
+  end
+
+  context "for an asset attached only to a superseded edition" do
```

Comment:
Is it worth drying up the first two and second two contexts? E.g. for the first two

```rb
%w[unpublished superseded].each do |state|
  context "for an asset attached only to a #{state} edition" do
      let(:edition) { create(:"#{state}_publication") }
  [...]
  end
end
```

Reply:
That seems like a good idea.  Have made this change.